src/asynk/async_worker.rs

```diff
@@ -193,8 +228,31 @@ impl<'a> AsyncWorkerTest<'a> {
                 .await
             {
                 Ok(Some(task)) => {
-                    self.sleep_params.maybe_reset_sleep_period();
-                    self.run(task).await?
+                    let actual_task: Box<dyn AsyncRunnable> =
+                        serde_json::from_value(task.metadata.clone()).unwrap();
+
+                    // check if task is scheduled or not
+                    match actual_task.cron() {
```

Comment:
the same comments as for the main `execute_task`

Reply:
[comments suggestions](https://github.com/ayrat555/fang/pull/71/commits/4eb86a2e2a6a2e33153736332de9191e4a162d56)